vpr/src/place/place.cpp

```diff
@@ -1746,6 +1901,33 @@ static e_move_result try_swap(const t_annealing_state* state,
     return move_outcome;
 }
 
+static bool is_cube_bb(const e_place_bounding_box_mode place_bb_mode,
+                       const RRGraphView& rr_graph) {
+    bool cube_bb;
+    const int number_layers = g_vpr_ctx.device().grid.get_num_layers();
```

Comment:
Does `place_bb_mode` change during a placer run? If not, it's best to calculate and hoist something like `g_vpr_ctx.placement().use_cube_bb` as soon as possible instead of having this fn and the extra parameters in placer fns.

If `place_bb_mode` can change while the placer runs (I don't think it does? seems to come from `placer_opts`), it would be better to precompute `inter_layer_...` (looks like it's a property of the RRG) and then expose `is_cube_bb` as a global inline/constexpr fn.

Reply:
You are correct; it doesn't change during placement. The issue arises when the `cube_bb` type is set to `auto`; in this case, I iterate over the RR graph to ensure all inter-layer connections are from OPINs (I don't rely on the architecture file to be able to support custom RR graph). To address this, I moved the `cube_bb` to the placement context and set it at the beginning of placement. Do you have any suggestions for an alternative approach?

Reply:
I think that's OK and matches Fahri's suggestion (set as soon as possible, which in this case is at the start of placement).